classes/local/store/object_file_system.php

```diff
@@ -736,6 +736,11 @@ public function presigned_url_should_redirect($contenthash) {
             return true;
         }
 
+        // If file MIME type is whitelisted we do not generate a presigned_url for it.
```

Comment:
I think we are doing more db calls than we need to, for this logic and the logic below around file size

Is there any way to infer the file type from the current http request without doing this db lookup? If not can we do 1 db lookup to grab the record and then use that for both bits of logic below?

Reply:
I ran into the same frustration, I ended up trying to do it later in the process once the Content-Type that was to be sent to the user was known and inferred it from that, then threw an exception which bubbled up and could be caught to safely abort the redirect.

This worked fine for the pre-signed test page but not for SCORM, possibly because I think (but didn't get around to fixing) that by the time I was doing it, Content-Type was mangled to include a suffix for indicating UTF-8 if it was otherwise text/html. But I ran out of time I could spend on it :(